internal/subscription/validation.go

```diff
@@ -114,78 +155,206 @@ func GetValidatedModelForUpdate(ctx context.Context, req *metrov1.Subscription)
 		}
 	}
+
 	return nil
 }
 
-func getValidatedModel(ctx context.Context, req *metrov1.Subscription) (*Model, error) {
+func validateSubscriptionName(ctx context.Context, m *Model, req *metrov1.Subscription) error {
 	// validate and extract the subscription fields from the name
 	p, s, err := extractSubscriptionMetaAndValidate(ctx, req.GetName())
 	if err != nil {
-		return nil, merror.Newf(merror.InvalidArgument, "Invalid [subscriptions] name: (name=%s)", req.Name)
+		return err
 	}
 
-	// get validated topic details
-	topicName, err := validateTopicName(ctx, req.GetTopic())
-	if err != nil {
-		return nil, merror.Newf(merror.InvalidArgument, "Invalid [subscriptions] topic: (topic=%s)", req.GetTopic())
+	m.Name = req.GetName()
+	m.ExtractedSubscriptionProjectID = p
+	m.ExtractedSubscriptionName = s
+
+	return nil
+}
+
+func validateTopicName(ctx context.Context, m *Model, req *metrov1.Subscription) error {
+	name := req.Topic
+	if strings.HasSuffix(name, topic.RetryTopicSuffix) {
+		return ErrInvalidTopicName
 	}
 
-	m := &Model{
-		Name:                           req.GetName(),
-		Topic:                          topicName,
-		Labels:                         req.GetLabels(),
-		ExtractedSubscriptionName:      s,
-		ExtractedSubscriptionProjectID: p,
+	p, t, err := topic.ExtractTopicMetaAndValidate(ctx, name)
+	if err != nil {
+		return err
 	}
 
-	return m, nil
+	m.Topic = name
+	m.ExtractedTopicProjectID = p
+	m.ExtractedTopicName = t
+
+	return nil
 }
 
-func validateTopicName(_ context.Context, name string) (string, error) {
-	if strings.HasSuffix(name, topic.RetryTopicSuffix) {
-		err := fmt.Errorf("subscription topic name cannot end with " + topic.RetryTopicSuffix)
-		return "", err
-	}
+func validateLabels(_ context.Context, m *Model, req *metrov1.Subscription) error {
+	// TODO: add validations on labels
+	labels := req.GetLabels()
 
-	return name, nil
+	m.Labels = labels
+	return nil
 }
 
-func validatePushConfig(_ context.Context, config *metrov1.PushConfig) (string, error) {
+func validatePushConfig(_ context.Context, m *Model, req *metrov1.Subscription) error {
+	config := req.GetPushConfig()
+
 	if config != nil {
+		// validate the endpoint
 		urlEndpoint := config.PushEndpoint
 		_, err := url.ParseRequestURI(urlEndpoint)
 		if err != nil {
-			return "", err
+			return ErrInvalidPushEndpointURL
+		}
+
+		pushAttr := config.GetAttributes()
+		var creds *credentials.Model
+
+		// check creds and encrypt if present
+		if pushAttr != nil {
+			var username, password string
+			if u, ok := pushAttr[attributeUsername]; ok {
+				if strings.Trim(u, " ") == "" {
+					return ErrInvalidPushEndpointUsername
+				}
+				username = u
+			}
+
+			if p, ok := pushAttr[attributePassword]; ok {
+				if strings.Trim(p, " ") == "" {
+					return ErrInvalidPushEndpointPassword
+				}
+				password = p
+			}
+
+			delete(pushAttr, attributeUsername)
```

Comment:
@rutul25 this should partial fix the creds being present in attributes, you will still need to have a way to fix the exisiting data models.